.github/workflows/typescript-type-generation.yml

```diff
@@ -38,7 +38,7 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          ref: ${{ github.event.inputs.branch || github.event.pull_request.head.ref || github.ref }}
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
This change will fail when the workflow is triggered on demand using workflow_dispatch

Reply:
```suggestion
          ref: ${{ inputs.branch || github.event.pull_request.head.ref || github.ref }}
```